autoprotocol/protocol.py

```diff
@@ -6618,11 +6641,14 @@ def location_helper(source, destination, volume, method):
 
         # if one tip is true then there's a locations list
         if locations:
+            source_transports = locations[0]["transports"]
+            if mode is None:
+                mode = LiquidHandle.builders.mode(source_transports, mode)
```

Comment:
I don't know how useful calling `LiquidHandle.builders.mode(source_transports, None)` would be, if `mode` is not provided. I'd argue that `mode` argument for this builder is unnecessary

Reply:
so true. removed `if mode is None:`